### PR TITLE
Rendering: Debug views

### PR DESCRIPTION
**Pull request by sp0lsh**
[F5] Lit
[F6] Unlit
[F7] Wireframe
[F8] Debug Normals

### CODE REVIEW

File: PolyEngine/RenderingDevice/OpenGL/Src/GLShaderProgram.hpp [hunk 1]
```diff
@@ -66,5 +66,9 @@ namespace Poly {
 		std::map<String, OutputInfo> Outputs;
 		GLuint ProgramHandle;
 		EnumArray<String, eShaderUnitType> ShaderCode;
+
+		String VertexProgramPath;
+		String GeometryProgramPath;
+		String FragmentProgramPath;
```
Comment (MuniuDev) [inline PolyEngine/RenderingDevice/OpenGL/Src/GLShaderProgram.hpp]: As stated in previous review, this fields should be removed.

File: PolyEngine/RenderingDevice/OpenGL/Src/GLWorldRendering.cpp [hunk 2]
```diff
@@ -40,62 +38,171 @@ void GLRenderingDevice::RenderWorld(World* world)
...
+			case eRenderingModeType::DEBUG_NORMALS:
+				RenderNormals(world, rect, cameraCmp);
+				break;
 
-		glDepthMask(GL_FALSE);
-		
-		glEnable(GL_BLEND);
+			case eRenderingModeType::LIT:
+			default:
+				RenderLit(world, rect, cameraCmp);
```
Comment (MuniuDev) [inline PolyEngine/RenderingDevice/OpenGL/Src/GLWorldRendering.cpp]: It's better to invoke assertion when enum is of unknown value.

File: PolyEngine/RenderingDevice/OpenGL/Src/GLWorldRendering.cpp [hunk 3]
```diff
@@ -40,62 +38,171 @@ void GLRenderingDevice::RenderWorld(World* world)
...
+	// Render text
+	GeometryRenderingPasses[eGeometryRenderPassType::TEXT_2D]->Run(world, cameraCmp, rect);
+
+	if (post && post->UseFgShader)
+		PostprocessRenderingPasses[ePostprocessRenderPassType::FOREGROUND]->Run(world, cameraCmp, rect);
+	else
+		PostprocessRenderingPasses[ePostprocessRenderPassType::FOREGROUND_LIGHT]->Run(world, cameraCmp, rect);
+
+	// Draw debug normals
+	if (gDebugConfig.DebugNormalsFlag)
```
Comment (MuniuDev) [inline PolyEngine/RenderingDevice/OpenGL/Src/GLWorldRendering.cpp]: We should consider moving this to CameraComponent as well.

Comment (MuniuDev) [reply PolyEngine/RenderingDevice/OpenGL/Src/GLWorldRendering.cpp]: My bad, this is already in camera component. Remove it from config then.

File: PolyEngine/RenderingDevice/OpenGL/Src/GLShaderProgram.hpp (hunk 1 above)
Comment (sp0lsh) [reply PolyEngine/RenderingDevice/OpenGL/Src/GLShaderProgram.hpp]: done

File: PolyEngine/RenderingDevice/OpenGL/Src/GLWorldRendering.cpp (hunk 2 above)
Comment (sp0lsh) [reply PolyEngine/RenderingDevice/OpenGL/Src/GLWorldRendering.cpp]: done

File: PolyEngine/RenderingDevice/OpenGL/Src/GLWorldRendering.cpp (hunk 3 above)
Comment (sp0lsh) [reply PolyEngine/RenderingDevice/OpenGL/Src/GLWorldRendering.cpp]: Debug rendering modes are already in camera, now removed from config and removed these calls

**Pull request merged**